### PR TITLE
Naive attempt to add https active checks

### PR DESCRIPTION
**Pull request by gaetanfl**
It's just an attempt to add https active checks.

I'm not sure about the type of failure to report.

### CODE REVIEW

**Comment by jeremyjpj0916** [comment]
This should probably be merged to Kong:next right? Def looking forward to https health checking with kong!

**Comment by Tieske** [comment]
@gaetanfl any chance you can add some tests to this code?

**Comment by gaetanfl** [comment]
@Tieske I'm not familair at all with tests I can give it a try next week.I saw this line in travis output which led me to think it will be difficult
` ./configure --prefix=$OPENRESTY_PREFIX --without-http_ssl_module -j$JOBS > build.log 2>&1`
Is there any reason for this `--without-http_ssl_module` ? Or it's nothing to be worried about ?

**Comment by gaetanfl** [comment]
@Tieske I'va added test but I have a failure which doesn't seem related to my work.
May I have some help with that ?

**Comment by gaetanfl** [comment]
@Tieske it was a race condition with prove -j2 and same bound ports so now it should be good for review

**Comment by Tieske** [comment]
@gaetanfl Thx. I noticed a lot of changes for just port numbers that greatly complicate the diffs/commits can that be reverted? So we can keep a clean history, and only see the real changes?

**Comment by gaetanfl** [comment]
I will create another PR for the test race condition fix

**Comment by gaetanfl** [comment]
PR #12 was created

**Comment by gaetanfl** [comment]
@Tieske I will rebase the branch on master when #12 will be merged to please travis

File: lib/resty/healthcheck.lua [hunk 1]
```diff
@@ -1189,7 +1195,7 @@ function _M.new(opts)
 
   assert(self.name, "required option 'name' is missing")
   assert(self.shm_name, "required option 'shm_name' is missing")
-  assert(({ http = true, tcp = true })[self.type], "type can only be 'http' " ..
+  assert(({ http = true, tcp = true, https = true })[self.type], "type can only be 'http(s)' " ..
```
[the remark this answers is not in the document]
Comment (gaetanfl) [reply lib/resty/healthcheck.lua]: Sure. It's done

**Comment by hishamhm** [comment]
@gaetanfl looks good! thank you for the update and for the ping!

**Comment by hishamhm** [comment]
@gaetanfl merged, thank you!!

**Pull request merged**